### ISSUE TITLE
What Postgres directory?

### ISSUE THREAD

**Issue by corepay** [furstenheim/is_jsonb_valid#4]
I would like to use this extension.  I am confused by this Testing and Installation instruction though:

My case in an ubuntu Postgres with apt-get.  

You say to do make install in the project where Postgres is located....I have several projects in different folders (node) running to the postrges server.  I don't get it.

Is it here where i am supposed to clone and make install?

![image](https://user-images.githubusercontent.com/48992923/57570472-551f0980-73d0-11e9-9e2e-110e892edc15.png)

Or here?

`$ ps auxw |  grep postgres | grep -- -D`
```
postgres 18277  0.0  2.5 319988 25204 ?        S    May10   0:01 /usr/lib/postgresql/10/bin/postgres -D /var/lib/postgresql/10/main -c config_file=/etc/postgresql/10/main/postgresql.conf
```

So in whatever directory I should 1. git clone then 2. make install && make installcheck.  I assume this will create a library or bundle.  Then you say from postgres to `CREATE EXTENSION is_jsonb_valid;`  I assume this will automatically find the library if I built it in right folder....?

Sorry if I am bothering you, found this library through a few posts you promoted it for json schema validation.  Would help a lot if you added a few details.  I appreciate your work.

**Comment by furstenheim** [comment]
Hi, I'm sorry if the readme is a bit confusing or lacking.

You should clone `is_jsonb_valid` in a different folder. Anywhere you want in your filesystem. Once you've cloned *this* project and cd it into it. You can do `make install && make installcheck`, this basically means executing the Makefile of the project.

About your last question: `I assume this will automatically find the library`. I think this [blog post](http://big-elephants.com/2015-10/writing-postgres-extensions-part-i/) from Manuel Kniep explains it better:

> Every PostgreSQL installation from 9.1 onwards provides a build infrastructure for extensions called PGXS, allowing extensions to be easily built against an already-installed server. Most of the environment variables needed to build an extension are setup in pg_config and can simply be reused.

So basically, if postgres was correctly installed, pg_config should return all the necessary environment variables (In this case saying that postgres is installed in /var/lib/postgresql/10/main). This will allow `make install` to create the required binary for the extension.

Improvements to the README are welcome ;)

**Comment by corepay** [comment]
Thank you for the response.  Let me give it a go and if I can reason through will see if I can add anything of value.

- Cheers

**Comment by furstenheim** [comment]
I'll close the issue. If you have more problems don't hesitate on opening a new issue.
